### ISSUE TITLE
label 정규화 실험 결과  (2023/04/18)

### ISSUE THREAD

**Issue by yunjinchoidev** [boostcamp-5th-NLP05/level1_semantictextsimilarity-nlp-05#9]
label의 점수를 정규화하여 학습하였을 때와 raw한 label로 학습을 진행하였고,
각각 30 epoch 씩 학습을 진행하였을 때 정규화한 결과가 성능이 조금 더 잘 나온 것을 확인했습니다.

label의 정규화 과정은 skeleton의 setup 함수에서 진행해야합니다.

```python
def setup(self, stage='fit'):
        if stage == 'fit':
            # 학습 데이터와 검증 데이터셋을 호출합니다
            train_data = pd.read_csv(self.train_path)
            val_data = pd.read_csv(self.dev_path)
            
            # 추가해야할 부분!
             t_label = []
             for label in train_data['label']:
                 t_label.append((label - 2.5) / 2.5)
             train_data['label'] = t_label
            
             v_label = []
             for label in val_data['label']:
                 v_label.append((label - 2.5) / 2.5)
             val_data['label'] = v_label
            # 추가해야할 부분 끝
            
            # 학습데이터 준비
            train_inputs, train_targets = self.preprocessing(train_data)

            # 검증데이터 준비
            val_inputs, val_targets = self.preprocessing(val_data)

            # train 데이터만 shuffle을 적용해줍니다, 필요하다면 val, test 데이터에도 shuffle을 적용할 수 있습니다
            self.train_dataset = Dataset(train_inputs, train_targets)
            self.val_dataset = Dataset(val_inputs, val_targets)
        else:
            # 평가데이터 준비
            test_data = pd.read_csv(self.test_path)
            test_inputs, test_targets = self.preprocessing(test_data)
            self.test_dataset = Dataset(test_inputs, test_targets)

            predict_data = pd.read_csv(self.predict_path)
            predict_inputs, predict_targets = self.preprocessing(predict_data)
            self.predict_dataset = Dataset(predict_inputs, [])

```

위 모델을 사용할 시 결과가 -1과 1 사이로 보통 나오기 때문에
inference 과정에서는 다시 0~5 범위로 정규화해주는 과정이 필요합니다.

```python
predictions = list(round(max(min((float(i) + 1) * 2.5, 5), 0), 1) for i in torch.cat(predictions))
```
